internal/controller/spectrumxrailpoolconfig_host_flows_controller.go

```
@@ -70,65 +101,449 @@ func NewSpectrumXRailPoolConfigHostFlowsReconciler(
+	}
+
+	if !rpc.DeletionTimestamp.IsZero() {
+		for _, rt := range rpc.Spec.RailTopology {
+			if err := r.deleteRailTopologyResources(ctx, rpc.Namespace, rt.Name); err != nil {
+				log.Error(err, "failed to delete rail topology resources", "rail topology", rt)
+				return ctrl.Result{}, err
+			}
+		}
+		patch := client.MergeFrom(rpc.DeepCopy())
+		controllerutil.RemoveFinalizer(rpc, finalizerName)
+		return ctrl.Result{}, r.Client.Patch(ctx, rpc, patch)
```

golangci-lint (2.11.3)

[error] QF1008: could remove embedded field "Client" from selector

(staticcheck)